src/hooks/useGetUserRewards.ts

```diff
@@ -0,0 +1,24 @@
+import { useFetcherWithSign } from "../utils/fetcher"
+import useSWR from "swr"
+import useGuild from "../components/[guild]/hooks/useGuild"
+import useUser from "../components/[guild]/hooks/useUser"
+
+export const useGetUserRewards = () => {
+  const { id: guildId } = useGuild()
+  const { id: userId } = useUser()
+
+  const fetcherWithSign = useFetcherWithSign()
+  const fetcher = (key: string) =>
```

Comment:
fetcher is a pretty general name (in fact, we have a `fetcher.ts` file in this repo), so I'd suggest you to find another name for this function, e.g. `fetchUsersRewards`